include/picongpu/fields/LaserPhysics.hpp

```diff
@@ -17,83 +17,198 @@
+            using LaserFunctor = T_LaserFunctor;
+
+            PMACC_CASSERT_MSG(
+                __LaserPlaneSizeInSuperCell_y_must_be_less_or_equal_than_SuperCellSize_y,
+                LaserPlaneSizeInSuperCell::y::value <= SuperCellSize::y::value
+            );
+
+            constexpr uint32_t planeSize = pmacc::math::CT::volume< LaserPlaneSizeInSuperCell >::type::value;
+            constexpr uint32_t numWorkers = T_numWorkers;
+
+            const uint32_t workerIdx = threadIdx.x;
+
+            // offset of the superCell (in cells, without any guards) to the origin of the local domain
+
+            DataSpace< simDim > localSuperCellOffset = DataSpace< simDim >( blockIdx );
+
+            // add not handled supercells from LaserFunctor::Unitless::initPlaneY
+            localSuperCellOffset.y() += LaserFunctor::Unitless::initPlaneY / SuperCellSize::y::value;
+
+            uint32_t cellOffsetInSuperCellFromInitPlaneY = LaserFunctor::Unitless::initPlaneY % SuperCellSize::y::value;
+
+            auto accLaserFunctor = laserFunctor(
+                acc,
+                localSuperCellOffset,
+                mappings::threads::WorkerCfg< numWorkers >{ workerIdx }
+            );
+
+            mappings::threads::ForEachIdx<
+                mappings::threads::IdxConfig<
+                    planeSize,
+                    numWorkers
+                >
+            > { workerIdx }(
+                [&](
+                    uint32_t const linearIdx,
+                    uint32_t const
+                )
+                {
+                    /* cell index within the superCell */
+                    DataSpace< simDim > cellIdxInSuperCell = DataSpaceOperations< simDim >::template map< LaserPlaneSizeInSuperCell >( linearIdx );
+                    cellIdxInSuperCell.y() += cellOffsetInSuperCellFromInitPlaneY;
+
+                    accLaserFunctor( acc, cellIdxInSuperCell );
+                }
+            );
+        }
+    };
+
+    /** Laser init in a single xz plane */
+    struct LaserPhysics
     {
-        float3_X elong;
-        float_X phase = 0.0;
-
-        elong = laserProfile::laserLongitudinal(currentStep,
-                                                phase);
-
-        const SubGrid<simDim>& subGrid = Environment<simDim>::get().SubGrid();
-
-        const DataSpace<simDim> globalCellOffset(subGrid.getLocalDomain().offset);
-        const DataSpace<simDim> halfSimSize(subGrid.getGlobalDomain().size / 2);
-
-        DataSpace<simDim> centeredOrigin(globalCellOffset - halfSimSize);
-        return LaserManipulator(elong,
-                                centeredOrigin,
-                                phase);
-    }
-
-private:
-
-    GridLayout<simDim> m_layout;
-};
-}
+        void operator()(uint32_t currentStep) const
+        {
+            /* The laser can be initialized in the plane of the first cell or
+             * any later x-y plane inside the simulation. Initializing the
```

Comment:
Question:
Is it not a x-z plane with a specific y coordinate/cell-value?

Reply:
yes must be `x-z`